### PR TITLE
Feature - add error states on the registration page

### PR DESCRIPTION
**Pull request by rikhilrai**
**A couple of examples:**

Screenshot 2021-01-05 at 21 46 07

Shame about the layout of the password input boxes here. Any thoughts on how to fix that? Maybe make a container around both boxes and align the inputs within that container?

Screenshot 2021-01-05 at 21 46 19

### CODE REVIEW

File: app/src/pages/loginPage/loginPage.tsx [hunk 1]
```diff
@@ -41,7 +41,9 @@ export const LoginPage = (props: ILoginPageProps): React.ReactElement => {
         await asyncSleep(1000);
         // eslint-disable-next-line no-console
         console.log('email', email && email.toLowerCase(), 'password', password);
+        history.navigate('/domains');
```
Comment (rikhilrai) [inline app/src/pages/loginPage/loginPage.tsx:44]: go to home page on successful login lol

File: app/src/pages/loginPage/loginPage.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ export const LoginPage = (props: ILoginPageProps): React.ReactElement => {
       hasErrors = true;
     }
     if (password && password.length < 8) {
-      setPasswordError('Use 8 or more characters');
+      setPasswordError('Wrong password. Try again or click Forgot password to reset it');
```
Comment (rikhilrai) [inline app/src/pages/loginPage/loginPage.tsx:35]: previous error didn't really make sense. Not sure if this error makes sense here anymore either cos we're not calling the backend lol. Maybe this is what the TODO on line 46 should return?

File: app/src/pages/registrationPage/registrationPage.tsx [hunk 3]
```diff
@@ -12,23 +12,100 @@ export interface IRegistrationPageProps {
...
   const onRegisterClicked = async (): Promise<void> => {
-    setIsLoading(true);
-    try {
-      asyncSleep(1000);
-      // eslint-disable-next-line no-console
-      console.log('email', email, 'password', password);
-    } catch (error) {
-      console.error('error', error);
+    let hasErrors = false;
+    let hasPasswordError = false;
```
Comment (rikhilrai) [inline app/src/pages/registrationPage/registrationPage.tsx:28]: had to add this additional flag cos we don't want both password boxes to show an error at the same time (idea taken from google's page) 
also we don't want the second password error to overwrite the first one 🤷

File: app/src/pages/registrationPage/registrationPage.tsx [hunk 4]
```diff
@@ -12,23 +12,100 @@ export interface IRegistrationPageProps {
...
 
   const onRegisterClicked = async (): Promise<void> => {
-    setIsLoading(true);
-    try {
-      asyncSleep(1000);
-      // eslint-disable-next-line no-console
-      console.log('email', email, 'password', password);
-    } catch (error) {
-      console.error('error', error);
+    let hasErrors = false;
```
Comment (rikhilrai) [inline app/src/pages/registrationPage/registrationPage.tsx]: can think of a better name for this? `hasNonPasswordRelatedErrors`?

File: app/src/pages/registrationPage/registrationPage.tsx [hunk 5]
```diff
@@ -12,23 +12,100 @@ export interface IRegistrationPageProps {
...
   const [firstName, setFirstName] = React.useState<string | null>(null);
+  const [firstNameError, setFirstNameError] = React.useState<string | null>(null);
   const [lastName, setLastName] = React.useState<string | null>(null);
+  const [lastNameError, setLastNameError] = React.useState<string | null>(null);
   const [email, setEmail] = React.useState<string | null>(null);
+  const [emailError, setEmailError] = React.useState<string | null>(null);
   const [password, setPassword] = React.useState<string | null>(null);
-  const [retypedPassword, setRetypedPassword] = React.useState<string | null>(null);
+  const [passwordError, setPasswordError] = React.useState<string | null>(null);
+  const [confirmPassword, setConfirmPassword] = React.useState<string | null>(null);
```
Comment (rikhilrai) [inline app/src/pages/registrationPage/registrationPage.tsx]: don't particularly like this name, but not too bad. any suggestions?

Comment (krishan711) [reply app/src/pages/registrationPage/registrationPage.tsx]: maybe `confirmedPassword`? tough one lol

File: app/src/pages/registrationPage/registrationPage.tsx (hunk 4 above)
Comment (krishan711) [reply app/src/pages/registrationPage/registrationPage.tsx]: ye maybe `hasNonPasswordError`?

**Review by krishan711 — APPROVED** [review]

**Pull request merged**